xtra-libp2p-rollover/src/maker.rs

```diff
@@ -137,50 +149,61 @@ where
             }
         };
 
-        // In case we fail to accept/reject some proposals might never get cleaned up. Given that
-        // the taker will retry this should not do much harm because we will replace the proposal
-        // with a new one. This is acceptable for now.
-        self.pending_protocols
-            .insert(order_id, (framed, peer, (from_event_id, from_complete_fee)));
-    }
+        if !self.is_accepting_rollovers {
+            emit_rejected(order_id, &self.executor).await;
 
-    async fn handle(&mut self, msg: Accept) -> Result<()> {
-        let Accept {
-            order_id,
-            tx_fee_rate,
-            long_funding_rate,
-            short_funding_rate,
-        } = msg;
+            let mut tasks = Tasks::default();
+            tasks.add_fallible(
+                async move {
+                    framed
+                        .send(ListenerMessage::Decision(Decision::Reject(Reject {
+                            order_id,
+                        })))
+                        .await
+                },
+                move |e| async move {
+                    tracing::debug!(%order_id, "Failed to send reject rollover to the taker: {e:#}")
+                },
+            );
+            self.protocol_tasks.insert(order_id, tasks);
+
+            return;
+        }
 
         fn next_rollover_span(parent: &tracing::Span) -> tracing::Span {
             tracing::debug_span!(parent: parent, "next rollover message")
         }
 
-        let (mut framed, _, from_params) =
-            self.pending_protocols.remove(&order_id).with_context(|| {
-                format!("No active protocol for {order_id} when accepting rollover")
-            })?;
-
         let mut tasks = Tasks::default();
         tasks.add_fallible(
             {
                 let executor = self.executor.clone();
                 let oracle = self.oracle.clone();
+                let rates =  self.rates.clone();
                 let oracle_pk = self.oracle_pk;
                 let n_payouts = self.n_payouts;
                 async move {
+                    let Rates {
```

Comment:
I suppose you're implying that because we removed the intermediate confirmation step on the maker for rolling over, we don't check that the `automaker` is up before rolling over. And if it is up we understand that the offer must be up-to-date.

I would argue that the current state of affairs of assuming that the offer is up-to-date simply because the `automaker` is up is not much safer than not checking anything at all. Nevertheless, I do think we should introduce a safety mechanism. But I would recommend doing in a separate PR, since this one is quite big already and the situation is not significantly worse with this work.

Reply:
> I suppose you're implying that because we removed the intermediate confirmation step on the maker for rolling over, we don't check that the automaker is up before rolling over. And if it is up we understand that the offer must be up-to-date.

yes, this was what I thought :) 

I think a follow-up PR is fine given the size of this PR. Can you make sure to create a ticket for that once this is merged please?

Reply:
Done: https://github.com/itchysats/itchysats/issues/2531.